doc/user/content/integrations/http-api.md

```diff
@@ -0,0 +1,173 @@
+---
+title: "Connect to Materialize via HTTP"
+description: "How to use Materialize via HTTP"
+menu:
+  main:
+    parent: "integrations"
+    weight: 5
+    name: "HTTP API"
+---
+
+You can access Materialize through its "session-less" HTTP API, reachable at
+`https://<MZ host address>/api/sql`.
+
+## Details
+
+### General semantics
+
+The API:
+
+- Requires username/password authentication, just as connecting via `psql`.
+  Materialize provides you the username and password upon setting up your
+  account.
+- Requires you must provide the entirety of your request. The API does not
+  provide session-like semantics, so there is no way to e.g. interactively use
+  transactions.
+- Ceases to process requests upon encountering the first error.
+- Does not support statements whose semantics rely on sessions or whose state is
```

Comment:
What about explicitly listing all unsupported statements here instead of trying to define what indeterminate means? I'm also surprised `DECLARE` isn't supported because...I think it should work fine?

Reply:
That requires folks making changes to statements/plans/execute responses to be aware of the fact that these docs need to be updated, which means they're almost certain to fall out of date. Being descriptive (rather than prescriptive) is ultimately less ambiguous.

Reply:
Also `DeclaredCursor` has always been prohibited via this API; I didn't dig into the historical reasons why.

Reply:
This sentence is trying to inform users of statements they cannot execute. I think it needs to be changed to be more clear. For example, *I* am currently unable to read this and know which SQL statement types aren't supported beyond the ones in the examples.

I agree it can get out of date if we specify a list here. But I think that's a less bad problem than trying to describe the general types of things that won't work. This should be clear enough that a non-employee with light familiarity with SQL and zero knowledge of postgres can read it and know what statements it doesn't support.

Reply:
Sure--I hedged this by saying that the partial set of unsupported statements are the full list we currently know.